cmp/cmp.go

```diff
@@ -38,32 +42,40 @@ func Float64Slice(f1, f2 []float64, tolerance float64) bool {
 	sort.Float64s(f1s)
 	sort.Float64s(f2s)
 	for i := range f1s {
-		if !Float64(f1s[i], f2s[i], tolerance) {
+		if !Float64(f1s[i], f2s[i], tolerance, bitTolerance) {
 			return false
 		}
 	}
 	return true
 }
 
 // Float64 compares two floats to see if they are within the given tolerance.
-func Float64(f1, f2, tolerance float64) bool {
-	if math.IsInf(f1, 1) {
-		return math.IsInf(f2, 1)
-	}
-	if math.IsInf(f2, 1) {
-		return math.IsInf(f1, 1)
+func Float64(f1, f2, tolerance float64, bitTolerance int64) bool {
+
+	// handle infinity
+	if math.IsInf(f1, 0) || math.IsInf(f2, 0) {
+		return math.IsInf(f1, -1) == math.IsInf(f2, -1) &&
+			math.IsInf(f1, 1) == math.IsInf(f2, 1)
 	}
-	if math.IsInf(f1, -1) {
-		return math.IsInf(f2, -1)
+
+	// -0.0 exist but -0.0 == 0.0 is true
+	if f1 == 0 || f2 == 0 {
+		return math.Abs(f2 - f1) < tolerance
 	}
-	if math.IsInf(f2, -1) {
-		return math.IsInf(f1, -1)
+
+	i1 := int64(math.Float64bits(f1))
+	i2 := int64(math.Float64bits(f2))
+	d := i2 - i1
+
+	if d < 0 {
+		return d > -bitTolerance
+	} else {
+		return d < bitTolerance
```

Comment:
Drop the else branch, it's unnecessary and can cause a bit of confusion when quickly reading the code.